nntrainer/tensor/var_grad.cpp

```diff
@@ -18,15 +18,15 @@
 
 namespace nntrainer {
 
-Var_Grad::Var_Grad(const TensorDim &dim, bool ng, bool alloc_now_,
-                   const std::string &name) :
+Var_Grad::Var_Grad(const TensorDim &dim, const Tensor::Initializer init,
+                   bool ng, bool alloc_now_, const std::string &name) :
   dim(dim),
   need_gradient(ng),
   alloc_now(alloc_now_),
   name(name) {
-  var = std::make_shared<Tensor>(dim, alloc_now);
+  var = std::make_shared<Tensor>(dim, alloc_now, init);
   if (need_gradient)
-    grad = std::make_shared<Tensor>(dim, alloc_now);
+    grad = std::make_shared<Tensor>(dim, alloc_now, Tensor::Initializer::ZEROS);
```

Comment:
if we allocate this to zero here, does this apply for the every iteration?

Reply:
No, initializer (as the name suggests) initializes only upon allocation by default.
Tensor::initialize() can be called again to re-initialize the variable.

Reply:
It is currently set to zero to keep the original behavior. Updating the behavior of when the gradient is zeroed will be another PR.

Reply:
Cool